packages/alfa-dom/src/node.ts

```diff
@@ -226,8 +226,22 @@ export abstract class Node
     yield* this.descendants();
   }
 
-  public equals(value: unknown): value is this {
-    return value === this;
+  protected _structurallyEquals(value: unknown): value is this {
+    return (
+      value instanceof Node &&
+      // We can't use Sequence#equals because it is oblivious to structural equality and will just call item.equals
+      Array.from(
+        this.children({ nested: true, composed: true })
+      ).every((thisChild, idx) =>
+        thisChild._structurallyEquals(
+          Array.from(value.children({ nested: true, composed: true }))[idx]
```

Comment:
Child lists aren't memoized so this will go through the work of constructing the child list for `value` for every child of `this`.